doc/programming_guide/sound.rst

```diff
@@ -0,0 +1,724 @@
+.. _Wave: https://en.wikipedia.org/wiki/WAV
+.. _MP3: https://en.wikipedia.org/wiki/MP3
+
+.. _Audacity: https://www.audacityteam.org/
+.. _FFmpeg: https://ffmpeg.org/
+
+.. _PyGame CE: https://pyga.me/
+.. _SDL2: https://www.libsdl.org/
+
+.. _pyglet media guide: https://pyglet.readthedocs.io/en/latest/programming_guide/media.html
+.. _pyglet's guide to supported media types: https://pyglet.readthedocs.io/en/latest/programming_guide/media.html#supported-media-types
+.. _pyglet_audio_drivers: https://pyglet.readthedocs.io/en/latest/programming_guide/media.html#choosing-the-audio-driver
+
+.. _sound:
+
+Sound
+=====
+
+This page will help you get started by covering the essentials of sound.
+
+In addition each section's concepts, there may also be links to example
+code and documentation.
+
+#. :ref:`sound-why-important`
+#. :ref:`sound-basics`
+
+   * :ref:`sound-basics-loading`
+   * :ref:`sound-basics-playing`
+   * :ref:`sound-basics-stopping`
+
+#. :ref:`sound-loading-modes`
+#. :ref:`sound-advanced-playback`
+#. :ref:`sound-compat`
+#. :ref:`sound-other-libraries` (for advanced users)
+
+.. rubric:: I'm Impatient!
+
+Users who want to skip to example code should consult the following:
+
+#. :ref:`sound_demo`
+#. :ref:`sound_speed_demo`
+#. :ref:`music_control_demo`
+#. :ref:`Platformer Tutorial - Step 7 - Add Coins And Sound <platformer_part_seven>`
+
+   #. :ref:`platformer_part_seven_loading_sounds`
+   #. :ref:`platformer_part_seven_playing_sounds`
+
+.. _sound-why-important:
+
+Why Is Sound Important?
+-----------------------
+
+Sound helps players make sense of what they see.
+
+This is about far more than the game being immersive or cool. For
+example, have you ever run into one of these common problems?
+
+* Danger you never knew was there
+* A character whose reaction seemed unexpected or out of place
+* Items or abilities which appeared similar, but were very different
+* An unclear warning or confirmation dialog
+
+How much progress did it cost you? A few minutes? The whole playthrough?
+More importantly, how did you feel? You probably didn't want to keep
+playing.
+
+You can use sound to prevent moments like these. In each example above,
+the right audio can provide the information players need for the game
+to feel fair.
+
+.. _sound-basics:
+
+Sound Basics
+------------
+
+.. _sound-basics-loading:
+
+Loading Sounds
+^^^^^^^^^^^^^^
+
+Before you can play a sound, you need to load its data into memory.
+
+Arcade provides two ways to do this. Both accept the same arguments and
+return an :py:class:`arcade.Sound` instance.
+
+The easiest way is :py:func:`arcade.load_sound`::
```

Comment:
Could we update this (and the others) to use this method of listing code samples?

```
.. code-block:: python

    print("Hello World!")
```